lib/heap_sort.rb

```diff
@@ -1,8 +1,53 @@
-
-
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity: O(nlogn)
+# >> creating max heap is O(nlogn)
+# >> removing from heap is done n times, each time is O(log n) --> O(nlogn)
+#
+# Space Complexity: O(1)
+# >> sorting in-place
+
 def heap_sort(list)
```

Comment:
👍 Really cool how you got this working with O(1) space complexity using the original array as the heap store.  Well done.